src/test/java/io/vertx/core/impl/DeploymentManagerTest.java

```diff
@@ -0,0 +1,20 @@
+package io.vertx.core.impl;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import org.junit.Test;
+
+import org.junit.Assert;
+
+public class DeploymentManagerTest {
+
+  @Test
+  public void extractClasspathTest() throws MalformedURLException, URISyntaxException {
+    for(URL cpURL : DeploymentManager.extractClasspath(getClass().getClassLoader())) {
+      Assert.assertTrue(cpURL + " is not a valid classpath entry", new File(cpURL.toURI()).exists());
+    }
```

Comment:
Add a test with an exploded classpath, it may behave differently.

Reply:
Done, good catch, this resulted really in a problem :)